### PR TITLE
[perf, dist] feat: add zero2 in fsdp1 and use_orig_params configurable

### PR DESCRIPTION
**Pull request by zhtao303**
### What does this PR do?
1、Enable `ShardingStrategy._HYBRID_SHARD_ZERO2` in parallelize_model_fsdp1
2、Enable `use_orig_params` to configure
> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `misc`, `ci`, `config`, `docs`, `data`, `dist`, `omni`, `logging`, `model`, `optim`, `ckpt`, `release`, `task`, `perf`, `ops`, `parallel`
  - If this PR involves multiple modules, separate them with `,` like `[ci, data, model]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][parallel, model] feat: dynamic batching`

### Test
1、When memory is sufficient, the `ShardingStrategy.SHARD_GRAD_OP` or `ShardingStrategy._HYBRID_SHARD_ZERO2` configuration can be enabled, preventing parameters from being resharded after the `forward`, thus avoiding the `all_gather` operation before the `backward`.We conducted experiments using a shallower `qwen2-vl` on these two configurations, achieved a 14% performance improvement.
image
image
Note that specific performance improvements depend on the model itself.

2、We noticed that the `use_orig_params` parameter is hardcoded to `true`. While it demonstrates advantages in various aspects, such as debugging and grouping optimizers, setting `use_orig_params=False` still has its uses if extreme training performance is desired. Therefore, we recommend making this parameter configurable.
We also conducted experiments on shallower qwen2vl, and found that `using_params=false` resulted in faster training speed.
image
image

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

**Comment by Luosuu** [comment]
Thank you for contribution! But I have a question.

Since PyTorch official is deprecating FSDP1, we are also planning to deprecate FSDP1 in the future when PyTorch finally deprecates it. Could you please why you are still using FSDP1, especially about why FSDP2 cannot satisfy your need? (because you are using NPU or what?)

**Comment by Luosuu** [comment]
Your linting is failing please reformat your file with ruff / precommit

**Comment by zhtao303** [comment]
> Thank you for contribution! But I have a question.
> 
> Since PyTorch official is deprecating FSDP1, we are also planning to deprecate FSDP1 in the future when PyTorch finally deprecates it. Could you please why you are still using FSDP1, especially about why FSDP2 cannot satisfy your need? (because you are using NPU or what?)



> Thank you for contribution! But I have a question.
> 
> Since PyTorch official is deprecating FSDP1, we are also planning to deprecate FSDP1 in the future when PyTorch finally deprecates it. Could you please why you are still using FSDP1, especially about why FSDP2 cannot satisfy your need? (because you are using NPU or what?)

Thank you for your question.
I haven't encountered any scenario where FSDP1 is strictly required. It's just   while reading the VeOmni source code, I noticed that when using FSDP1, one must use `ShardingStrategy.FULL_SHARD`. If I have sufficient memory, I might be willing to trade extra memory for faster training speed.
Btw, even in FSDP2, a behavior similar to FSDP1's `ShardingStrategy.SHARD_GRAD_OP` (i.e., ZeRO-2) can be achieved by setting `reshard_after_forward=False`. Therefore, I believe it's still worthwhile to support ZeRO-2 (SHARD_GRAD_OP) in FSDP1 for completeness and flexibility.
Happy to discuss further    (๑•̀ㅂ•́)و✧

**Comment by Luosuu** [comment]
Sure. Thanks for your contribution.
